airflow/contrib/operators/gcp_function_operator.py

```diff
@@ -0,0 +1,563 @@
+            if not self.body[SOURCE_UPLOAD_URL]:
+                self.upload_function = True
+            else:
+                raise AirflowException("Only one of '{}' in body or '{}' argument "
+                                       "allowed. Found both."
+                                       .format(SOURCE_UPLOAD_URL, ZIP_PATH))
+
+    def _is_archive_url_and_zip_path(self):
+        if SOURCE_ARCHIVE_URL in self.body and self.zip_path:
+            raise AirflowException("Only one of '{}' in body or '{}' argument "
+                                   "allowed. Found both."
+                                   .format(SOURCE_ARCHIVE_URL, ZIP_PATH))
+
+    def should_upload_function(self):
+        if self.upload_function is None:
+            raise AirflowException('validate() method has to be invoked before '
+                                   'should_upload_function')
+        return self.upload_function
+
+    def preprocess_body(self):
+        self._is_archive_url_and_zip_path()
+        self._is_upload_url_and_zip_path()
+        self._is_upload_url_and_no_zip_path()
+        if self.upload_function is None:
+            self.upload_function = False
+
+
+FUNCTION_NAME_PATTERN = '^projects/[^/]+/locations/[^/]+/functions/[^/]+$'
+FUNCTION_NAME_COMPILED_PATTERN = re.compile(FUNCTION_NAME_PATTERN)
+
+
+class GcfFunctionDeleteOperator(BaseOperator):
+    """
+    Delete a function with specified name from Google Cloud Functions.
+
+    :param name: A fully-qualified function name, matching
+        the pattern: `^projects/[^/]+/locations/[^/]+/functions/[^/]+$`
+    :type name: str
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud Platform.
+    :type gcp_conn_id: str
+    :param api_version: Version of the API used (for example v1).
+    :type api_version: str
+    """
+
+    @apply_defaults
+    def __init__(self,
+                 name,
+                 gcp_conn_id='google_cloud_default',
+                 api_version='v1',
+                 *args, **kwargs):
+        self.name = name
+        self.gcp_conn_id = gcp_conn_id
+        self.api_version = api_version
+        self._validate_inputs()
+        self.hook = GcfHook(gcp_conn_id=self.gcp_conn_id, api_version=self.api_version)
+        super(GcfFunctionDeleteOperator, self).__init__(*args, **kwargs)
+
+    def _validate_inputs(self):
+        if not self.name:
+            raise AttributeError('Empty parameter: name')
+        else:
+            pattern = FUNCTION_NAME_COMPILED_PATTERN
+            if not pattern.match(self.name):
+                raise AttributeError(
+                    'Parameter name must match pattern: {}'.format(FUNCTION_NAME_PATTERN))
+
+    def execute(self, context):
+        try:
+            return self.hook.delete_function(self.name)
+        except HttpError as e:
+            status = e.resp.status
+            if status == 404:
+                self.log.warning('The function does not exist in this project')
```

Comment:
I might put this as info-level. Airflow operators should be designed for idempotence, so a missing function that you are trying to delete should not be surprising. Up to you.

Reply:
Info is indeed better.